src/query/storage/m3/storage.go

```diff
@@ -163,6 +163,11 @@ func (s *m3storage) Fetch(
 		fetchResult.Metadata.Resolutions = resolutions
 	}
 
+	if options.IncludeExemplars {
+		fetchResult.Metadata.ExemplarsList = fetchResult.ExemplarsList
+		fetchResult.ExemplarsList = nil
```

Comment:
I think better than doing it here would be to feed the FetchOptions through to the `SeriesIteratorsToFetchResult` method and let that deal with setting exemplars?